express-notes-api/server.js

```diff
@@ -103,21 +90,19 @@ app.put('/api/notes/:id', (req, res) => {
   } else if (dataParse.notes[entryId] === undefined) {
     const errorMessageNoId = { error: `Cannot find note with id ${entryId}` };
     res.status(404).json(errorMessageNoId);
-  } else if (dataParse.notes[entryId].id === entryId) {
+  } else {
     updateEntry.id = entryId;
     dataParse.notes[entryId] = updateEntry;
-    res.status(204).json(updateEntry);
     const dataStringify = JSON.stringify(dataParse, null, 2);
     writeFile('data.json', dataStringify)
       .then((success) => {
+        res.status(204).json(updateEntry);
         console.log('Note was successfully updated');
       })
       .catch((error) => {
-        console.log(error.message);
+        res.status(500).json(error.errorMessage);
```

Comment:
See above about logging and process.exit. Also, `error` probably does not have a property named `errorMessage`. There should be one named `message`, though.